Amazon.IonDotnet.Tests/Internals/BinaryReaderTimestampTest.cs

```diff
@@ -34,23 +34,39 @@ public void Date_2000_11_20_8_20_15_Unknown()
             ReaderTimestampCommon.Date_2000_11_20_8_20_15_Unknown(reader);
         }
 
-        [DataRow("2008-12-23T23:00:01.123+07:00")]
+        [DataRow("2008T", DateTimeKind.Unspecified, 0, "2008T")]
+        [DataRow("2008-02T", DateTimeKind.Unspecified, 0, "2008-02T")]
+        [DataRow("2008-02-03T", DateTimeKind.Unspecified, 0, "2008-02-03T")]
+        [DataRow("2008-12-23T23:22:00-00:00", DateTimeKind.Unspecified, 0, "2008-12-23T23:22:00")]
+        [DataRow("2008-12-23T23:22:33", DateTimeKind.Unspecified, 0, "2008-12-23T23:22:33")]
```

Comment:
`2008-12-23T23:22:33` is not a valid Ion timestamp and should raise an error because it includes time but not a local offset. See http://amzn.github.io/ion-docs/docs/spec.html#timestamp

Reply:
Fixed the bug and the tests - updated the PR.
(This commit specifically: https://github.com/amzn/ion-dotnet/pull/100/commits/a7709f4a12a54dc6b0851c0fb2c07c85c259504f)